js/h5p.js

```diff
@@ -224,10 +224,16 @@ H5P.init = function (target) {
         instance.getCurrentState instanceof Function ||
         typeof instance.getCurrentState === 'function')) {
 
-      var saveTimer, save = function () {
+      var saveTimer, lastSavedState, save = function () {
         var state = instance.getCurrentState();
-        if (state !== undefined) {
+
+        //compare last state with current state
+        var stateChanged = H5P.eq(state, lastSavedState, null, null);
```

Comment:
The "object's internal state" abstraction is really leaky here.  Those last two null arguments need to be moved out of the calling interface and into "implementation details that are invisible to the caller"